tinync/README.md

```diff
@@ -0,0 +1,78 @@
+# Tinync
+Tinync is a simplified version of [nc](https://en.wikipedia.org/wiki/Netcat), which aims to demonstrate the usage of coroutine macros and their effects.
+
+## Usage
+### Declaring a Coroutine Function
+At the very beginning, a coroutine function should be declared first, coroutine function is part of program that will be executed simultaneously with other coroutines. All coroutines should be specified with the macro `cr_proto`, following shows the example:
+```cpp
+static void cr_proto(coroutine_name, int x, float y)
+{
+    /* coroutine body */
+}
+```
+Programmers are free to modify the argument list as they need.
+
+### Controlling Macros
+With in the coroutine, there are several macros for controlling the behavior of coroutine, following are list of them:
+* `cr_begin` aims to initiate the context of coroutine when the coroutine is first invoked, or resume previous execution for the rest of invocations. This macro should be placed before other cotrolling macros except `cr_local`.
+* `cr_end` ends up a coroutine that will mark the status of coroutine as finished, which could be detected outside to ensure whether a coroutine finisihed its job or not.
+* `cr_wait` is used for waiting a condition happened. Once the condition it is waiting for has not happened yet, it will pause the current coroutine and switch to another.
+* `cr_exit` not only yields a coroutine but also updates its state with given state.
+* `cr_sys` is a wrapper of `cr_wait` which performs waiting on system calls and other functions that return -1 and set `errno`.
+* `cr_local` is a marker for programmers to recognize a variable related to coroutine easily.
+In `tinync.c`, we can see the combinations of these macros:
+```cpp
+static void cr_proto(stdin_loop, byte_queue_t *out)
+{
+    /* b and r are variables used in coroutine whose
+     * value will be preserved across pauses.
+     */
+    cr_local uint9_t b;
```

Comment:
uint9_t ?

Reply:
Sorry for the mistyped, I will fix that.